backend/lib/config/gardener.js

```diff
@@ -55,20 +125,43 @@ function getEnvironmentVariable (env, name, type) {
       return value
   }
 }
+function getValueFromEnvironment (env, envVar, type) {
+  const value = env[envVar]
+  return convertValue(value, type)
+}
+
+function getValueFromFile (filePath, type) {
+  try {
+    if (!fs.existsSync(filePath)) {
+      return undefined
+    }
+    const fileContent = fs.readFileSync(filePath, 'utf8')
+    return convertValue(fileContent, type)
+  } catch (error) {
+    /* ignore */
+  }
+}
+
+function getValueFromEnvironmentOrFile (env, envVar, filePath, type) {
+  const value = getValueFromEnvironment(env, envVar, type)
+  if (value !== undefined) {
+    return value
+  }
+
+  if (filePath) {
+    return getValueFromFile(filePath, type)
+  }
+}
 
 module.exports = {
-  assignEnvironmentVariables (config, env) {
-    _.forEach(environmentVariableDefinitions, (path, name) => {
-      let type = 'String'
-      if (_.isPlainObject(path)) {
-        type = path.type
-        path = path.path
-      }
-      const value = getEnvironmentVariable(env, name, type)
-      if (value) {
-        _.set(config, path, value)
+  setConfigFromEnvOrFiles (config, env) {
```

Comment:
rename to `extendConfigFromEnvironmentAndFileSystem`

Reply:
✅  
renamed to `assignConfigFromEnvironmentAndFileSystem`